### PR TITLE
Merge patch-174-173 to master

### LINKED ISSUE #173: Node Internal Module Errors
In cmd\powershell shell terminals - observed internal module errors in the natives directory of node when attempt to perform "npm start"

`node -v` returns **v14.2.0**
`npm -v` returns **6.14.5**

---

I handled the error by updating several node packages, and will be creating a Pull Request for this issue shortly

### LINKED ISSUE #174: 'npm audit' is not working and provides generic error message
In terminal (while in the project directory scope), type and run `npm audit` (or `npm audit fix`)

With default or properly configured npm registry - the operation should run an audit against the local dependency packages (and attempt to fix it if you specified `fix`).   

Although this issue is observed with an improperly message indicating the configured npm registry is either not responding or is potential not able to handle audit requests, this is a project specific issue.

The issue relates to not specifying the full file extension `.git` and\or not providing the protocol.

https://github.com/cblanc/sws_gathers/blob/e2dc987d6d2aa4eacd7a36cd7a744866241915c2/package.json#L22

---

to resolve this, modify url to be `git+https://github.com/cblanc/sws_gathers.git` in the packages.json file.  

I will include the resolving of this issue in a Pull Request that I will create shortly.

### PR DESCRIPTION
**Pull request by brcaswell**
Changes relating to npm start (handling babel-brunch compiling and development environment observations)
- updates node package versions to resolve #173
  - ensure devDependencies are defined as such
- modify url in packages.json to resolve #174; 
- add "test-win" to npm script running, (relates to terminal \ host environment compatibility, as `npm test` will fail in `cmd` and `powershell`) will have to explicitly do `npm run test-win` to run script.
  - and modify env assignment for resiliency

resolves #173 
resolves #174

### CODE REVIEW

**Review by brcaswell — COMMENTED** [review]
Some changes I need to make after reviewing the diff in this PR

File: package.json [hunk 1]
```diff
@@ -19,60 +25,61 @@
...
-    "lodash": "~4.17.16",
-    "moment": "~2.11.2",
+    "lodash": "^4.17.20",
+    "moment": "^2.27.0",
     "mongoose": "~4.4.6",
-    "morgan": "~1.6.1",
+    "morgan": "^1.10.0",
     "newrelic": "~1.22.1",
     "perfect-scrollbar": "~0.6.10",
+    "popper.js": "^1.16.1",
```
[the remark this answers is not in the document]
Comment (brcaswell) [reply package.json:55]: dependency of Bootstrapper for hover over behavior I believe.

[the remark this answers is not in the document]
Comment (brcaswell) [reply package.json:55]: example doc https://getbootstrap.com/docs/4.1/components/popovers/#example

note that it is npm indicated as a missing dependency of the package, which is why it was included.

**Comment by brcaswell** [comment]
> @cblanc can you check why this fails? Seems I don't have the permissions to do so.

If I had to guess, I would think the failing check relates to steam package.  As it seems to fail it's prepare script.   

I looked into the package's source and project a bit, and I don't know why that package author is doing svn checkouts against github projects to begin with, but it would seem having a svn tool is an unspecified requirement

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (brcaswell) [reply package.json:55]: you're right though - I'm going to remove it.  

It is outside the scope of this PR and I have no issue writeup to merit the change.

File: package.json [hunk 2]
```diff
@@ -19,60 +25,60 @@
...
-    "discord.js": "^11.1.0",
-    "express": "~4.15.5",
-    "express-handlebars": "~2.0.1",
+    "discord.js": "^12.3.1",
+    "express": "^4.17.1",
+    "express-handlebars": "^5.1.0",
     "extend": "~3.0.0",
     "howler": "~1.1.28",
-    "javascript-brunch": ">= 1.0 < 1.8",
     "javascript-state-machine": "~2.3.5",
```
Comment (brcaswell) [inline package.json:47]: this is a devDependency as well

**Comment by cblanc** [comment]
Hey - i'd love to get back to this but I have no mental energy outside of work. I can only really offer some high level and probably less than useful pointers

There are 3 problems with this codebase - all of my own making

1. CI needs to be enabled. Although this project has some unit testing, it's not hooked up to a CI system
2. The project needs to be in a docker container. This will simplify local and production setup across envs. Something like [this](https://github.com/ideal-postcodes/paf-api/blob/master/Dockerfile) would suffice. Very simple, small + efficient. MongoDB has a very good stock docker image
3. Tests should also be completely dockerised. Like [here](https://github.com/ideal-postcodes/postcodes.io/blob/master/Makefile#L37. Again tests can be run in 1 command on any platform including CI

This solves a bunch of problems:

1. Anyone can make changes and test very quickly. Only a docker daemon is required (all available on MS, MacOS and Linux)
2. You can merge PRs quickly & with more confidence
3. Dev and prod env parity. No more works on this machine but not that
4. Dependency upgrades can be automated because CI
5. 1 command setup makes simple e2e tests very easy to setup if you need more confidence
6. This app needs to be moved off heroku because Mongodb service is shutting down. Would like to move to a low end VPS. Will be even cheaper and have better performance. Deployment and updates as simple as heroku if the app can be shipped as a docker container

**Comment by Absurdon** [comment]
Actually im planing on dockerizing this thing anyways. When it's docker we can just put it on jirikis machine so no need to pay an additional hoster. 
Sadly i dont have the time either. If anyone has go for it.
I think november was the deadline for this.

**Comment by Absurdon** [comment]
when i did dockerization i merged this one in

**Pull request closed without merging**